src/cryptonote_basic/cryptonote_format_utils.cpp

```diff
@@ -543,9 +533,8 @@ namespace cryptonote
       CHECK_AND_NO_ASSERT_MES_L1(r, false, "failed to deserialize extra field. extra = " << string_tools::buff_to_hex_nodelimer(std::string(reinterpret_cast<const char*>(tx_extra.data()), tx_extra.size())));
       tx_extra_fields.push_back(field);
 
-      std::ios_base::iostate state = iss.rdstate();
-      eof = (EOF == iss.peek());
-      iss.clear(state);
+      eof = ar.eof();
+      ar.clear_fail();
```

Comment:
The old version looked unnecessarily complex. I hope the new version works equivalently. Do you know which test verifies this?

Reply:
The proof functional tests might test this, at least one type of proof runs two serializations on the same buffer, concatenated.

Reply:
I'm going to add a unit test for this. The code here changes the intent slightly, although it doesn't matter because the only error type is EOF now.

After a refactor these two lines will be just `ar.eof()`. If this triggers an error in the stream type (which it can't for this type), then the reset/clear behavior needs to go in `ar.eof()` instead.

Reply:
And there'll definitely be a comment in header for `eof()` explaining this, in case an overload for `std::istream` is re-added for some reason (read from unencrypted file would be only use case).